src/main/java/dev/andrylat/vomelianchuk/card_number_operations/PaymentSystem.java

```diff
@@ -0,0 +1,47 @@
+package dev.andrylat.vomelianchuk.card_number_operations;
+
+import java.util.Optional;
+
+public enum PaymentSystem {
+    VISA(4),
+    MASTERCARD(51, 52, 53, 54, 55),
+    DINERS_CLUB(36, 38),
+    DISCOVER(6011, 65),
+    JCB(35),
+    AMERICAN_EXPRESS(34, 37);
+
+    private static final String ERROR_CANNOT_DETERMINE_PAYMENT_SYSTEM = "Payment System can't be determine";
+
+    private final int[] prefixes;
+
+    PaymentSystem(int... prefixes) {
+        this.prefixes = prefixes;
+    }
+
+    public int[] getPrefixes() {
+        return prefixes;
+    }
+
+    public static PaymentSystem determine(String cardNumber) throws WrongPaymentSystemException {
```

Comment:
throws WrongPaymentSystemException is not needed.

I will explain why:

Imagine situation, that this class is used in 46 different locations. and IN EACH LOCATION they have try-catch block to handle the exception. Everything is ok.
BUT: tomorrow you need to introduce NEW exception and add it to the throws statement. Could you imagine scope of changes? It is huge! Huge regression, huge potential chance to have a bug, etc. Don't use checked exceptions and throws statement.

Reply:
Fixed here: 7a87509